src/operators.jl

```diff
@@ -858,3 +855,26 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where U <: Integer) where T
 
     return new_vertex_ids
 end
+
+# special case for digraphs
+function merge_vertices!(g::AbstractGraph, vs::Vector{U} where U <: Integer)
+    @assert is_directed(g)
```

Comment:
Instead of an assert here, we should prefer `@traitfn merge_vertices!(g::AbstractGraph::IsDirected...)`.